src/private/files.ts

```diff
@@ -0,0 +1,160 @@
+     */
+    ownerDisplayName: string;
+    /**
+     * Sharepoint specific siteURL of the folder
+     */
+    siteUrl?: string;
+    /**
+     * Sharepoint specific serverRelativeUrl of the folder
+     */
+    serverRelativeUrl?: string;
+    /**
+     * Sharepoint specific libraryType of the folder
+     */
+    libraryType?: string;
+    /**
+     * Sharepoint specific accessType of the folder
+     */
+    accessType?: string;
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Gets a list of cloud storage folders added to the channel
+   * @param channelId ID of the channel whose cloud storage folders should be retrieved
+   * @param callback Callback that will be triggered post folders load
+   */
+  export function getCloudStorageFolders(
+    channelId: string,
+    callback: (error: SdkError, folders: CloudStorageFolder[]) => void,
+  ): void {
+    if (!channelId || channelId.length == 0) {
+      throw new Error('[files.getCloudStorageFolders] channelId name cannot be null or empty');
+    }
+    if (!callback) {
+      throw new Error('[files.getCloudStorageFolders] Callback cannot be null');
+    }
+
+    ensureInitialized();
+    sendMessageToParent('files.getCloudStorageFolders', [channelId], callback);
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Initiates the add cloud storage folder flow
+   * @param channelId ID of the channel to add cloud storage folder
+   * @param callback Callback that will be triggered post add folder flow is compelete
+   */
+  export function addCloudStorageFolder(
+    channelId: string,
+    callback: (error: SdkError, isFolderAdded: boolean, folders: CloudStorageFolder[]) => void,
+  ): void {
+    if (!channelId || channelId.length == 0) {
+      throw new Error('[files.addCloudStorageFolder] channelId name cannot be null or empty');
+    }
+    if (!callback) {
+      throw new Error('[files.addCloudStorageFolder] Callback cannot be null');
+    }
+
+    ensureInitialized();
+    sendMessageToParent('files.addCloudStorageFolder', [channelId], callback);
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Deletes a cloud storage folder from channel
+   * @param channelId ID of the channel where folder is to be deleted
+   * @param folderToDelete cloud storage folder to be deleted
+   * @param callback Callback that will be triggered post delete
+   */
+  export function deleteCloudStorageFolder(
+    channelId: string,
+    folderToDelete: CloudStorageFolder,
+    callback: (error: SdkError, isFolderDeleted: boolean) => void,
+  ): void {
+    if (!channelId) {
+      throw new Error('[files.deleteCloudStorageFolder] channelId name cannot be null or empty');
+    }
+    if (!folderToDelete) {
+      throw new Error('[files.deleteCloudStorageFolder] folderToDelete cannot be null or empty');
+    }
+    if (!callback) {
+      throw new Error('[files.deleteCloudStorageFolder] Callback cannot be null');
+    }
+
+    ensureInitialized();
```

Comment:
Please changes this to `ensureInitialized(FrameContexts.content)` since this API should never be called from any other frame context.